### ISSUE TITLE
breaking change in 0.19.10 when using absolute paths?

### ISSUE THREAD

**Issue by stefanocke** [systemjs/systemjs#1019]
Hello,

I have another Plunkr that breaks starting from 0.19.10: https://plnkr.co/edit/sxBv3xGemi7o8AuaQIcC?p=preview

Note: I am again not sure if this is a "legal" configuration, but for example https://github.com/mgechev/angular2-seed uses absolute paths in the development build.

```
<!-- Switch to systemjs_0.19.13.js here to see the breaking change in the Console. -->
    <script src="systemjs_0.19.9.js"></script>
    
    <script>System.config({
        paths: {
          'myapp/*' : '/*'
        },
        packages: {
              'myapp': {
                defaultExtension: 'js'
              }
        }
      })</script>
    
    
    <script>
     System.register("foobar/core", [], true, function(require, exports, module) {})
      System.import('foobar/core')
        .then(function(m){ console.log('Loaded foobar/core:'+m)})
        .catch(function (e) {
          console.error(e,'!!! ERROR !!!');
        });
    </script>
```

**Comment by guybedford** [comment]
@stefanocke thanks this is actually a valid regression, the js extension gets added to the import as it is within the myapp package (being the root package), but the js extension does not get added to the register statement, as register skips extensions.

A hacky workaround is to turn foobar into its own URL-schema so it no longer exists within the same root path of the myapp package with the default extension:

```javascript
paths: {
  'foobar/*': 'foobar:*'
}
```

So it is a wontfix, but if this is something that's going to be a problem for Angular workflows, we can potentially discuss backwards compatibility options (perhaps through the defaultJSExtensions flag)?

**Comment by stefanocke** [comment]
@guybedford thanks for your feedback.
Short answer: 
Wontfix is fine with me. My primary intention was to document the changed behavior if someone else comes across the same issue.

Long answer:
I don't think there is currently a real problem for angular2-seed. Actually, they are still using defaultJSExtensions and it works. I came across this issue when trying to move from defaultJSExtensions to package-style configuration.
I don't like the workaround that you suggested, since it has to be applied for every library (angular, rxjs, angular router and so on). My intention was to keep the SystemJS configuration as small as possible and just to say: "Use .js extension for my app modules but for nothing else."
Meanwhile I came to the conclusion that the best solution is to always have a real "top-level-package" for my app modules. That is, to have somethink like "src/app/" which is transpiled to "dist/app/", instead of just having "app/" transpiled to "dist/". By this, I always have a clear way to identify my app modules and there is no danger of accidentally normalizing foobar or other libraries in such a way that they seem to belong to the app package. I don't have to map the app package to root anymore in the paths config.
